tests/engines/xarray_engine/test_plugin.py

```diff
@@ -2,211 +2,141 @@
+INT_1D_COORDS = {"rows": np.arange(0, 16, dtype=np.int32)}
+
+INT_2D_DATA = np.arange(0, 32, dtype=np.int32).reshape(8, 4)
+INT_2D_ATTRS = {"description": "int 2D data var"}
+INT_2D_VAR = Variable(["rows", "cols"], INT_2D_DATA, INT_2D_ATTRS)
+INT_2D_COORDS = {"rows": np.arange(0, 8, dtype=np.int32), "cols": np.arange(0, 4, dtype=np.int32)}
+
+INT_1D_SHIFTED_ATTRS = {"description": "int 1d shifted data var"}
+INT_1D_SHIFTED_VAR = Variable(["rows"], INT_1D_DATA, INT_1D_SHIFTED_ATTRS)
+INT_1D_SHIFTED_COORDS = {"rows": np.arange(5, 21, dtype=np.int32)}
+
+INT_2D_SHIFTED_ATTRS = {"description": "int 2d shifted data var"}
+INT_2D_SHIFTED_VAR = Variable(["rows", "cols"], INT_2D_DATA, INT_2D_SHIFTED_ATTRS)
+INT_2D_SHIFTED_COORDS = {"rows": np.arange(4, 12, dtype=np.int32), "cols": np.arange(1, 5, dtype=np.int32)}
+
+DATETIME_1D_VAR_ATTRS = {"description": "datetime 1d data var"}
+DATETIME_1D_VAR = Variable(["rows"], INT_1D_DATA, DATETIME_1D_VAR_ATTRS)
+DATETIME_1D_COORDS = {"rows": np.arange(np.datetime64("2000-01-01"), np.datetime64("2000-01-17"), np.timedelta64(1, 'D'))}
+
+DS_ATTRS = {"description": "test dataset"}
+DATETIME_1D_DS_ATTRS = {**DS_ATTRS, "rows": {"time reference": np.datetime64("2000-01-01"), "freq": "D"}}
+
+SAMPLE_DATASETS = {
+    "simple_1d": xr.Dataset(data_vars={"data": INT_1D_VAR}, coords=INT_1D_COORDS, attrs=DS_ATTRS),
+    "simple_2d": xr.Dataset(data_vars={"data": INT_2D_VAR}, coords=INT_2D_COORDS, attrs=DS_ATTRS),
+    "1d_shifted": xr.Dataset(data_vars={"data": INT_1D_SHIFTED_VAR}, coords=INT_1D_SHIFTED_COORDS, attrs=DS_ATTRS),
+    "2d_shifted": xr.Dataset(data_vars={"data": INT_2D_SHIFTED_VAR}, coords=INT_2D_SHIFTED_COORDS, attrs=DS_ATTRS),
+    "1d_datetime": xr.Dataset(data_vars={"data": DATETIME_1D_VAR}, coords=DATETIME_1D_COORDS, attrs=DATETIME_1D_DS_ATTRS)
+}
+
+class TestTDBBackend:
+    engine = "tiledb"
+
+    def open(self, path):
+        with open_dataset(path, engine=self.engine) as ds:
+            ds_obj = ds
+        return ds_obj
+
+    def to_tiledb(self, dataset, path):
+        tdb_dims = []
+        coords = dataset.coords
+        coords_attrs = dict()
+        for name in coords:
+            if name in dataset.dims:
+                if dataset.attrs.get(name) is not None:
+                    coords_attrs[name] = dataset.attrs[name]
+                coord = coords[name]
+                dtype = coord.dtype
+                if dtype.kind not in 'iuM':
+                    raise NotImplementedError(f"TDB Arrays don't work yet with this dtype coord {dtype}")
+                min_value = coord.data[0]
+                max_value = coord.data[-1]
+                if dtype.kind == "M":
+                    domain = (1, len(coord.data))
+                    dtype = np.int32
+                else:
+                    domain = (min_value + 1, max_value + 1)
+                tdb_dim = tiledb.Dim(name=name, domain=domain, dtype=dtype)
+                tdb_dims.append(tdb_dim)
```

Comment:
It might be easier to map coordinate to a TileDB dimension and a TileDB attribute instead of trying to combine them into a single dimension. For example, if you had a float64 xarray coordinate `x = (-1.0, -0.5, 0.5, 1.0)` you would map that to a TileDB integer dimension with values `x=[0,1,2,3]` and a floating point TileDB attribute with value `x.data = (-1.0, -0.5, 0.5, 1.0)`.

Reply:
Working on this next